docs/resources/aws_vpc_endpoint.md

```diff
@@ -0,0 +1,106 @@
+---
+title: About the aws_vpc_endpoint Resource
+platform: aws
+---
+
+# aws\_vpc\_endpoint
+
+Use the `aws_vpc_endpoint` InSpec audit resource to test properties of a single specific AWS VPC Endpoint. 
+
+A VPC Endpoint is uniquely identified by the VPC Endpoint ID (e.g vpce-123456abcdef12345)
+
+## Syntax
+
+Ensure that a VPC Endpoint exists.
+
+    # Find a VPC Endpoint by ID
+    describe aws_vpc_endpoint('vpce-12345678987654321') do
+      it { should exist }
+    end
+    
+    # Hash syntax for ID
+    describe aws_vpc_endpoint(vpc_endpoint_id: 'vpce-12345678987654321') do
+      it { should exist }
+    end
+
+Ensure that a VPC Endpoint is available.
+
+    describe aws_vpc_endpoint('vpce-12345678987654321') do
+      its('state') { should be 'available' }
+    end
+
+Confirm that the route table configured to a VPC Endpoint is as expected.
+
+    describe aws_vpc_endpoint('vpce-12345678987654321') do
+      its('route_table_ids') { should include 'rtb-1234456123456abcde' }
+    end
+
+Confirm that the type of a VPC Endpoint is as expected.
+
+    describe aws_vpc_endpoint('vpce-12345678987654321') do
+      its('vpc_endpoint_type') { should be 'Gateway' }
+    end
+
+## Parameters
+
+This resource expects the VPC Endpoint ID as a parameter.
+
+See also the [AWS documentation on VPC Endpoints](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-endpoints.html).
+
+## Properties
+
+|Property           | Description|
+| ---               | --- |
+|vpc_endpoint_id       | The ID of the endpoint |
+|vpc_endpoint_type | One of "Interface", "Gateway" |
+|vpc_id           | The ID of the VPC in which the endpoint resides. |
+|state | State of the VPC Endpoint. One of "pendingacceptance", "pending", "available", "deleting", "deleted", "rejected", "failed", "expired" |
+|route_table_ids | The route table IDs for the Gateway type endpoint. |
+|subnet_ids | The subnet IDs for the Interface type endpoint. |
+|groups | The Security Groups for the Interface type endpoint. |
+| private_dns_enabled   | Boolean value for Private DNS enable status.                 |
+|network_interface_ids            | The Network Interface IDs for the Interface type endpoint. |
+|dns_entries | The DNS Entries for the VPC Endpoint. |
+|tags | The key/value combination of a tag assigned to the resource. |
+
+## Examples
+
+##### Ensure a VPC Endpoint is available.
+    describe aws_vpc_endpoint(vpc_endpoint_id: 'vpce-12345678987654321') do
+      its('state') { should eq 'available' }
```

Comment:
It would be nice to give an example of the `available?` method being used here.
eg.
```
    describe aws_vpc_endpoint(vpc_endpoint_id: 'vpce-12345678987654321') do
       it { should be_available }
    end
```

Reply:
In fact, it would be good to document the `?` methods under the matchers section similar to this: https://github.com/inspec/inspec-aws/blob/d14788549c2368bd4fd4d2ebbb6d608f9424152f/docs/resources/aws_s3_bucket_object.md#be_public

Reply:
Thanks for the suggestion. Added both in examples and matchers.